lms/static/scripts/frontend_apps/utils/dashboard/hooks.ts

```diff
@@ -0,0 +1,54 @@
+import { useCallback, useMemo } from 'preact/hooks';
+import { useLocation, useSearch } from 'wouter-preact';
+
+import { queryStringToRecord, recordToQueryString } from '../url';
+
+export type DashboardFilters = {
+  studentIds: string[];
+  assignmentIds: string[];
+  courseIds: string[];
+};
+
+export type UseDashboardQueryFilters = {
+  filters: DashboardFilters;
+  updateFilters: (filtersToUpdate: Partial<DashboardFilters>) => void;
+};
+
+/**
+ * Allows to track dashboard filters as a piece of state synced in the query string
```

Comment:
I think the first line of the JSDoc comment here could provide a better summary of what the hook does / returns. Something like:

```
Hook used to read and update the state of the dashboard filters.

The filter state is synchronized with query string parameters in the URL.
```